### PR TITLE
Add an NCover-compatible XML report output

### PR DESCRIPTION
**Pull request by jonjomckay**
I had to change the structure of the output coverage JSON to make this fit (so it's not compatible with previously-output coverage), but now people can use something like [ReportGenerator](https://github.com/danielpalme/ReportGenerator) to generate HTML reports :)

### CODE REVIEW

**Comment by lucaslorentz** [comment]
That's awesome! I will review it later. Thanks for contributing!

File: src/MiniCover/Instrumentation/Uninstrumenter.cs [hunk 1]
```diff
@@ -9,16 +9,16 @@ public static void Execute(InstrumentationResult result)
         {
             foreach (var assembly in result.Assemblies)
             {
-                if (File.Exists(assembly.BackupFile))
+                if (File.Exists(assembly.Value.BackupFile))
```
Comment (lucaslorentz) [inline src/MiniCover/Instrumentation/Uninstrumenter.cs]: You can avoid `.Value` by changing the foreach to `foreach (var assembly in result.Assemblies.Values)`

File: src/MiniCover/Instrumentation/Instrumenter.cs [hunk 2]
```diff
@@ -154,13 +157,17 @@ private void InstrumentAssembly(string assemblyFile)
 
                             var instructionId = ++id;
 
-                            result.AddInstruction(sourceRelativePath, new InstrumentedInstruction
+                            result.Assemblies[assemblyDefinition.Name.Name].AddInstruction(sourceRelativePath, new InstrumentedInstruction
```
Comment (lucaslorentz) [inline src/MiniCover/Instrumentation/Instrumenter.cs]: I would prefer that the method AddInstrumentedAssembly returns the InstrumentedAssembly object.

Then we don't need to access the assembly by dictionary key here.

**Comment by lucaslorentz** [comment]
I did some tests and it is completely functional and reports generated with ReportGenerator looks good.

No blocker in the comments.

Let me know if you would like to change something, otherwise will merge the way it is.

**Comment by jonjomckay** [comment]
Thanks for the feedback @lucaslorentz - I made the changes!

**Pull request merged**